src/resource/iri.internal.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { IriString } from "../interfaces";
+
+/**
+ * This function normalizes IRIs as managed by the server to ease accurate comparison.
+ * @param iri
+ * @hidden
+ */
+export function normalizeServerSideIri(iri: IriString): IriString {
```

Comment:
I don't really follow this comment, and it's not an accurate description of what the function does, right? i.e., shouldn't the function just be called `removeIriHashFragment(iri))`, and a detailed comment added to wherever this function is being called to explain why we need to call it in that context? (In other words, we _could_ consider adding a detailed comment to this function trying to explain use-cases for calling it, but that feels a bit like added use-case comments to a `String.trim()` function, so it doesn't feel right).

Reply:
My thinking was the following: this function intends to take a user-supplied IRI, and normalize it to compare it to the server "view" of the same IRI. At the present, it only strips the hash fragment, but I don't exclude that we discover additional bugs like this one that require to do additional normalization on the IRI. 

In other words, the name of the function here describes the context in which it should be used, rather than exactly what it does.